request/process.go

```diff
@@ -52,35 +73,21 @@ func Process() {
 	}
 }
 
-// Request length is the first 4 bytes in LittleEndian encoding on stdin
-func parseRequestLength() uint32 {
+// Request length is the first 4 bytes in LittleEndian encoding
+func parseRequestLength(input io.Reader) (uint32, error) {
 	var length uint32
-	if err := binary.Read(os.Stdin, binary.LittleEndian, &length); err != nil {
-		log.Error("Unable to parse the length of the browser request: ", err)
-		response.SendErrorAndExit(
```

Comment:
OK, provided this error still gets sent to the client somehow. I'm assuming you intend for this to happen in the caller now.

Reply:
yeah, this code is moved above to the lines 35-55 with no changes

Reply:
I thought so, but just wanted to double-check your intention was to keep it that way, in light of your comment about changing things so that Go errors were returned instead.

Reply:
I re-read my message and now I see how it is confusing... So confirming: yes, the point is to make this helper function return the normal go error, and make the caller send it to the browser extension.